maps/mp/isnipe/extras/antihardscope.gsc

```diff
@@ -2,35 +2,57 @@
 #include maps\mp\_utility;
 #include maps\mp\gametypes\_hud_util;
 
-EnableAntiHardScope(time)
-{
-    self endon( "disconnect" );
-    self endon( "death" );
+/**
+ * Initialize the antiHardScope variables on the level.
+ *
+ * The following settings are available to be set outside this script:
+ * - adsTime (default: 3): the time that a player is allowed to aim down sight.
+ */
+init() {
+    if (!isDefined(level.antiHardScope)) {
+        level.antiHardScope = [];
+    }
+
+    // Set settings to default when they are not set.
+    _setSettingIfNotExists("adsTime", 3);
+}
 
-    if( !isDefined( time ) || time < 0.05 )
-        time = 3;
+/**
+ * Set defaultValue in level.antiHardScope[property] if it does not exist.
+ *
+ * @param property: the setting property to act upon
+ * @param defaultValue: the default value to set for the property if not set
+ */
+_setSettingIfNotExists(property, defaultValue) {
+    if (!isDefined(level.antiCamp[property])) {
+        level.antiCamp[property] = defaultValue;
```

Comment:
This is putting the property in the wrong array. 
It gets put into the `level.antiCamp` array instead of the `level.antiHardScope` array, which results in the adsTime variable not being read by the code at line 48.